### DISCUSSION TITLE
Need help using icon packs in my project

### DISCUSSION THREAD

**Discussion by pacjo** [breezy-weather/breezy-weather#602] [Q&A]
Hi, I love your app and since it has integration with gadgetbrodge I'm using it to get weather data easily from intents.

I wanted to add support for multiple icon packs, but I don't know where to start. Can you point me to some relevant documentation or even implementation files?

As a sidenote ability to enable gadgetbridge integration without having gadgetbridge installed would be nice.

**Comment by papjul** [comment]
What do you mean multiple icon packs? You want to mix multiple icon packs together (i.e. take the sun icon from pack 1, rain icon from pack 2, etc)? In that case, best is to extract icons you're interested in, and mix them into a new icon pack.

If you want to create your own icon pack for Breezy Weather, please check https://github.com/breezy-weather/breezy-weather-icon-packs/blob/main/README.md

**Comment by pacjo** [comment]
No, I'd like to have an ability to choose one icon pack from which I'll get all icons. I don't need multiple at one time.

Also I'm not looking into creating the pack for breezy weather, instead I want to utilize already existing icon packs in my project. I hope that's clear.

**Reply by papjul** [reply]
You will find icons of Breezy Weather in the drawable/ repo (beginning with `shortcut_` or `weather_`: https://github.com/breezy-weather/breezy-weather/tree/main/app/src/main/res/drawable

Pixel icon pack sources are here: https://github.com/WangDaYeeeeee/FullSizePixelIconProvider

Other icon packs designed by Wangdaye are not open source as far as I know.

**Reply by pacjo** [reply]
> You will find icons of Breezy Weather in the drawable...

Thanks, but that's what I'm using right now. I'm wondering if it's possible to use already existing icon packs like pixel one you provided in my app without manually including all resources. I want end user to be able to change icons without me having to update the app to include new resources.

**Reply by papjul** [reply]
I didn't work on that part specifically, but if you're looking for source code, you can just copy it from Breezy Weather source. Just make sure to comply with the LGPL-3.0-only license ;)

**Reply by pacjo** [reply]
That's the issue I'm facing. I'm looking for source code, but I can't find the part responsible for getting the resources from icon pack. License wont be an issue.

**Reply by papjul** [reply]
Manifest intents:
https://github.com/breezy-weather/breezy-weather/blob/main/app/src/main/AndroidManifest.xml#L46-L57

Theming: https://github.com/breezy-weather/breezy-weather/tree/main/app/src/main/java/org/breezyweather/theme/resource

**Reply by pacjo** [reply]
I'll try to piece something together, thanks
